primitives/io/src/batch_verifier.rs

```diff
@@ -155,12 +151,8 @@ impl BatchVerifier {
 
 				return false;
 			}
-
-			let (finished, cond_var) = &*pair;
-			let mut finished = finished.lock().expect("Locking can only fail when the mutex is poisoned; qed");
-			while !*finished {
-				finished = cond_var.wait(finished).expect("Waiting can only fail when the mutex waited on is poisoned; qed");
-			}
+			receiver.recv()
+				.expect("This can panic only if the spawned tasks is panicked. Spawned task does not panic; qed.");
```

Comment:
This is a claim that you can not verify. Either we ignore the error or print some warning on error.